deps/find_libpython.py

```diff
@@ -182,8 +182,9 @@ def candidate_names(suffix=SHLIB_SUFFIX):
     sysdata = dict(
         v=sys.version_info,
         # VERSION is X.Y in Linux/macOS and XY in Windows:
-        VERSION=(sysconfig.get_config_var("VERSION") or
-                 "{v.major}.{v.minor}".format(v=sys.version_info)),
+        VERSION=(sysconfig.get_python_version() or
+                 "{v.major}.{v.minor}".format(v=sys.version_info) or
+                 sysconfig.get_config_var("VERSION")),
```

Comment:
`sysconfig.get_config_var("VERSION")` never gets used since `"{v.major}.{v.minor}".format(v=sys.version_info)` is always truthy (non-empty string).

Also, reading https://docs.python.org/3/library/sysconfig.html#sysconfig.get_python_version, it looks like it's always X.Y while we need XY in Windows (guessing from the comment). I guess we just should have formatted this directly from `sys.version_info`.

Reply:
Hmm yeah it seems the comment says so, but how come the Windows CI passes?  Seems we tested this script in `test_build.jl`.

Reply:
`find_libpython.py` contains a bunch of fallbacks to make it as robust as possible. So the CI in PyCall may not be using this code path.

Also, since `find_libpython.py` is mainly maintained in PyJulia, it has a bit more tests there. So, in general, if you want to improve `find_libpython.py`, it's better to do this against PyJulia repository. I don't think this case is caught by the CI, though.